### ISSUE TITLE
Use updated PyTorch 0.4 tensor creation functions

### ISSUE THREAD

**Issue by jacobrgardner** [cornellius-gp/gpytorch#136]
Right now, we do the following kind of things to get new tensors:
```python
bar_randn = foo.new(foo.size(-2), foo.size(-1)).normal_()
bar_zeros = foo.new(size1, size2).zeros_()
bar_inds = foo.new(size1, size2).zeros_().long()
```
PyTorch 0.4.0 added the following ways to do this instead that I prefer:
```python
bar_randn = torch.randn_like(foo)
bar_zeros = foo.new_zeros(size1, size2)
bar_inds = foo.new_zeros(size1, size2, dtype=torch.long)
```
We could also just by carrying around a `device` string instead of a tensor_cls do:
```python
bar = torch.zeros(size1, size2, dtype=torch.long, device=device_str)
```

**Comment by Balandat** [comment]
I definitely like the second method better as well.

The `device` kwarg method is nice in that it can also take the actual `torch.device`, e.g. you can do something like

    bar = torch.zeros(size1, size2, dtype=foo.dtype, device=foo.device)

This makes sense (in particular if you have multiple GPUs...), but it's also pretty verbose since you're going to carry the device around everywhere.

**Comment by Balandat** [comment]
Also, let's not forget getting rid of `Variable` (#110) while at it (pass in `requires_grad` directly into the Tensor constructors if a gradient is needed)

**Comment by gpleiss** [comment]
LazyVariables should also get a `device` property, and we should get rid of the `tensor_cls` method.

**Comment by Balandat** [comment]
And they should also have a `dtype` property and `shape` property (which just wraps `size()`, as pytorch does). That way they become more similar to tensors, almost warranting a renaming....

**Comment by jacobrgardner** [comment]
They already have a `device` property -- I assume he means `dtype`.

**Comment by Balandat** [comment]
Yeah let's kill off `tensor_cls` and make sure LazyTensors implement both `dtype` and `device` properties in a way compatible to regular tensors. Both can be initialized from whatever base tensor / LazyTensor is passed into the constructor. We could even make these optional kwargs in the constructor, but I don't think that's necessary.
